runtime/queries/metricsview_aggregation.go

```diff
@@ -503,7 +505,9 @@ func (q *MetricsViewAggregation) buildMetricsAggregationSQL(mv *runtimev1.Metric
 		if err != nil {
 			return "", nil, err
 		}
-		havingClause = "HAVING " + havingClause
+		if strings.TrimSpace(havingClause) != "" {
+			havingClause = "HAVING " + havingClause
+		}
```

Comment:
It already checks that `q.Having != nil` – how can the it evaluate to an empty having clause if it's not nil?

Reply:
This is to gracefully handle cases with empty AND/OR or IN filters.